theano/tensor/tests/test_opt.py

```diff
@@ -5443,6 +5443,32 @@ def test3(self):
         assert len(divs) == 0
 
 
+def test_local_merge_alloc():
+    x = T.iscalar('x')
+    y = T.iscalar('y')
+    z = T.iscalar('z')
+    w = T.iscalar('w')
+    m = T.fscalar('m')
+    # case 1
+    # Alloc(Alloc(m, x, 1, 1, 1), x, y, z, w) -> Alloc(m, x, y, z, w)
+    output = T.alloc(T.alloc(m, 1, y, 1, 1), x, y, z, w)
+    f = theano.function([m, x, y, z, w], output)
+    topo = f.maker.fgraph.toposort()
```

Comment:
That test will fail if the optimization you are testing is not included, which is the case in FAST_COMPILE for instance.
Other tests in that file create a special mode that explicitly includes the necessary optimization, for instance:
```python
default_mode = theano.compile.mode.get_default_mode()
opt_mode = default_mode.including("local_...")
...
f = theano.function(..., mode=opt_mode)
...
```
Can you do that here? This should fix the failing test in Travis in FAST_COMPILE.